storage/remote/codec.go

```diff
@@ -277,8 +277,10 @@ func (c *concreteSeriesIterator) Err() error {
 	return nil
 }
 
-// validateLabelsAndMetricName validates the label names/values and metric names returned from remote read.
+// validateLabelsAndMetricName validates the label names/values and metric names returned from remote read,
+// also making sure that there are no labels with duplicate names
 func validateLabelsAndMetricName(ls labels.Labels) error {
+	existingLabelNames := make(map[string]bool, len(ls))
```

Comment:
Labels will be in order, so you only need to compare with the previous one.

Reply:
Nice catch. I didn't notice that `labelProtosToLabels` sorts the labels.

I updated the validation function and added a corresponding test case (as well as updating the tests to actually check that expectedErr matches err).